ice40/pack.cc

```diff
@@ -1125,14 +1140,20 @@ static void pack_special(Context *ctx)
 
                 if (net == nullptr)
                     continue;
+
                 if ((pi.name != ctx->id("RGB0")) && (pi.name != ctx->id("RGB1")) && (pi.name != ctx->id("RGB2")))
                     continue;
 
                 if (net->users.size() > 0)
-                    log_error("SB_RGBA_DRV port connected to more than just package pin !\n");
+                    log_error("SB_RGB_DRV/SB_RGBA_DRV port connected to more than just package pin !\n");
 
                 ctx->nets.erase(net->name);
             }
+
+            if (is_sb_rgb_drv(ctx, ci) && !ci->ledInfo.ledCurConnected)
```

Comment:
As far as I can tell, there is no guarantee that the SB_LED_DRV_CUR will be visited before the SB_RGB_DRV, so this could fail with some input netlists (probably depending on small changes to the Verilog affecting the order of cells)

Reply:
the cells are iterated in sorted (string index) order - i think.

Reply:
It's a bit more icky than that, they are sorted in IdString order, which will reflect the order in which IdStrings are created, ultimately reflecting the order in which cells were processed while reading the JSON. Yosys doesn't really guarantee the order of cells written in the JSON.

Reply:
Oh, I thought the initial IdStrings are created from constids.  If not, how do you suggest we handle this?

Reply:
That covers IdStrings used for build time constants like cell types, but not runtime things like cell names (which we sort by here). I would add another loop to deal with the CUR first.